File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    
    public int climbStairs(int n) {
        int[] mem = new int[50];
        mem[0] = 0; 
        mem[1] = 1;
        mem[2] = 2;
        mem[3] = 3;
        return climb(n , mem);
    }
    private int climb(int n, int[] mem){
        if (mem[n] > 0) return mem[n];
        mem[n - 1] = climb(n - 1, mem);
        mem[n - 2] = climb(n - 2, mem);
        return mem[n - 1] + mem[n - 2];
    }
}

File: LeetCode/Easy/0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n == 2: return 2
        if n == 1: return 1
        if not n: return 0
        steps = [0] * n
        steps[0], steps[1], steps[2] = 0, 1, 2
        for i in range(3, n):
            steps[i] = steps[i-2] + steps[i-1]
        return steps[n-1] + steps[n-2]

        